nx/lib/nx.ex

```diff
@@ -1440,6 +1440,176 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Creates a new tensor by repeating the input tensor
+  along the given axes.
+
+  If the `tensor` has less dimensions than the repetitions given,
+  the tensor will grow in dimensionality.
+
+  If the `tensor` has more dimensions than the repetitions given,
+  tiling is done from the rightmost dimensions (i.e. if the input
+  shape is `{1,2,3}` and `repetitions = [2]`, the result is the same
+  as if `repetitions = [1,1,2]`).
+
+  ## Examples
+
+      iex> a = Nx.tensor([0, 1, 2])
+      iex> Nx.tile(a, 2)
+      #Nx.Tensor<
+        s64[6]
+        [0, 1, 2, 0, 1, 2]
+      >
+      iex> Nx.tile(a, [1, 2])
+      #Nx.Tensor<
+        s64[1][6]
+        [
+          [0, 1, 2, 0, 1, 2]
+        ]
+      >
+      iex> Nx.tile(a, [2, 2])
+      #Nx.Tensor<
+        s64[2][6]
+        [
+          [0, 1, 2, 0, 1, 2],
+          [0, 1, 2, 0, 1, 2]
+        ]
+      >
+      iex> Nx.tile(a, [2, 1])
+      #Nx.Tensor<
+        s64[2][3]
+        [
+          [0, 1, 2],
+          [0, 1, 2]
+        ]
+      >
+      iex> Nx.tile(a, [2, 1, 2])
+      #Nx.Tensor<
+        s64[2][1][6]
+        [
+          [
+            [0, 1, 2, 0, 1, 2]
+          ],
+          [
+            [0, 1, 2, 0, 1, 2]
+          ]
+        ]
+      >
+
+      iex> b = Nx.tensor([[1,2],[3,4]])
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [1, 2],
+          [3, 4]
+        ]
+      >
+      iex> Nx.tile(b, 2)
+      #Nx.Tensor<
+        s64[2][4]
+        [
+          [1, 2, 1, 2],
+          [3, 4, 3, 4]
+        ]
+      >
+      iex> Nx.tile(b, [2, 1])
+      #Nx.Tensor<
+        s64[4][2]
+        [
+          [1, 2],
+          [3, 4],
+          [1, 2],
+          [3, 4]
+        ]
+      >
+      iex> Nx.tile(b, [1, 2])
+      #Nx.Tensor<
+        s64[2][4]
+        [
+          [1, 2, 1, 2],
+          [3, 4, 3, 4]
+        ]
+      >
+
+      iex> c = Nx.tensor([1,2,3,4])
+      #Nx.Tensor<
+        s64[4]
+        [1, 2, 3, 4]
+      >
+      iex> Nx.tile(c, [4,1])
+      #Nx.Tensor<
+        s64[4][4]
+        [
+          [1, 2, 3, 4],
+          [1, 2, 3, 4],
+          [1, 2, 3, 4],
+          [1, 2, 3, 4]
+        ]
+      >
+
+    ### Error cases
+
+    iex> Nx.tile(Nx.tensor([1,2]), 1.0)
+    ** (ArgumentError) repetitions must be either an integer or a list of integers. Got: 1.0
+
+    iex> Nx.tile(Nx.tensor([1,2]), [1, 1.0])
+    ** (ArgumentError) repetitions must be either an integer or a list of integers. Got: [1, 1.0]
+
+    iex> Nx.tile(Nx.tensor([1,2]), nil)
+    ** (ArgumentError) repetitions must be either an integer or a list of integers. Got: nil
+  """
+  @doc type: :shape
+  def tile(tensor, repetitions) do
+    %T{shape: old_shape} = tensor = tensor!(tensor)
+
+    invalid_repetitions_message =
+      "repetitions must be either an integer or a list of integers. Got: #{inspect(repetitions)}"
```

Comment:
This means you are always building the error message, even if you never raise.

Perhaps it is better to write it like this:

```elixir
unless valid_repetition?(repetitions) do
  raise ...
end
```